webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SecretApi.scala

```diff
@@ -0,0 +1,170 @@
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+package com.normation.rudder.rest.lift
+
+import com.normation.box._
+import com.normation.rudder.domain.secrets.Secret
+import com.normation.rudder.rest.ApiPath
+import com.normation.rudder.rest.ApiVersion
+import com.normation.rudder.rest.AuthzToken
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.rest.RestUtils
+import com.normation.rudder.rest.{SecretVaultApi => API}
+import com.normation.rudder.web.services.SecretService
+import net.liftweb.common._
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import net.liftweb.json.JArray
+import net.liftweb.json.JsonDSL._
+
+
+class SecretApi(
+    restExtractorService : RestExtractorService
+  , secretService        : SecretService
+) extends LiftApiModuleProvider[API] {
+
+  def schemas = API
+
+  def getLiftEndpoints(): List[LiftApiModule] = {
+    API.endpoints.map(
+      e => e match {
+        case API.GetSecrets   => GetSecrets
+        case API.AddSecret    => AddSecret
+        case API.UpdateSecret => UpdateSecret
+        case API.DeleteSecret => DeleteSecret
+      }
+    )
+  }
+
+  object GetSecrets extends LiftApiModule0 {
+    val schema        = API.GetSecrets
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action = schema.name
+
+      val res = for {
+        secrets <- secretService.getSecrets
+      } yield {
+        secrets.map(Secret.serializeSecret)
+      }
+
+      res.toBox match {
+        case Full(json) =>
+          RestUtils.toJsonResponse(None, JArray(json))
+        case eb: EmptyBox =>
+          val message = (eb ?~ s"Error when trying get secret from file").messageChain
+          RestUtils.toJsonError(None, message)(action, true)
+      }
+    }
+  }
+
+  object AddSecret extends LiftApiModule0 {
+    val schema        = API.AddSecret
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action = schema.name
+
+      val res = for {
+        secret <- restExtractorService.extractSecret(req)
+        _      <- secretService.addSecret(secret).toBox
+      } yield {
+        Secret.serializeSecret(secret)
+      }
+
+      res match {
```

Comment:
Use this function instead that will provide you everything you need, please use it at all function: 

```      RestUtils.response(restExtractor, "secrets", None)(res.toBox, req, "Error when trying to add a secret") ```